### PR TITLE
Fix potentially failing pkg-config commands in Allwmake

### LINKED ISSUE #190: Missing exit message in `Allwmake`
Due to the immediate exit
https://github.com/precice/openfoam-adapter/blob/a5c01a0993c34787c146c70368dd01cec37673e0/Allwmake#L2
 the `Allwmake` script 'exits' as follows:

```
Building the OpenFOAM-preCICE adapter...
```
in case the pkg-config is not discoverable. There is actually a message missing, which tells the user what went wrong.

### PR DESCRIPTION
**Pull request by DavidSCN**
Fixes #190

### CODE REVIEW

**Review by MakisH — APPROVED** [review]
Good addition, it covers yet another case in which things can go wrong.

I improved some minor aspects, please squash before merging (or revert).

File: Allwmake [hunk 1]
```diff
@@ -32,11 +32,24 @@ log "Building the OpenFOAM-preCICE adapter..."
...
 export ADAPTER_PREP_FLAGS
 export ADAPTER_TARGET_DIR
-ADAPTER_PKG_CONFIG_CFLAGS="$(pkg-config --silence-errors --cflags libprecice)"
+
+if pkg-config libprecice; then
+  ADAPTER_PKG_CONFIG_CFLAGS="$(pkg-config --silence-errors --cflags libprecice)"
+  ADAPTER_PKG_CONFIG_LIBS="$(pkg-config --silence-errors --libs libprecice)"
+else
+  log "WARNING: No configuration file 'libprecice.pc' was found in the PKG_CONFIG_PATH."
+  log "Check the preCICE documentation page 'Linking to preCICE' for more details, or proceed if you know what you are doing."
```
Comment (DavidSCN) [inline Allwmake:41]: Do you mean here by `know what you are doing' someone who exports all necessary paths?

Comment (MakisH) [reply Allwmake:41]: For example, yes. We want that the "regular" user uses pkg-config, but there are also other such ways one could install preCICE and make it discoverable.

Comment (DavidSCN) [reply Allwmake:41]: Alright. Is 'Linking to precice' the right place to look at/ is there the relevant information? or do we actually want to encourage people to export the `PKG_CONFIG_PATH`? Maybe we should add a few sentences in the adapter wiki and refer to it here?!

Comment (MakisH) [reply Allwmake:41]: We currently describe how to export the `PKG_CONFIG_PATH` in the "linking to preCICE" page: https://precice.org/installation-linking.html#make-and-scripts-in-general

I am not sure if we should use URLs in error messages, they tend to get outdated and forgotten.

Comment (DavidSCN) [reply Allwmake:41]: I wouldn't put the link here either. I'm just afraid that the description there is too developer oriented. I think the right place to look at for these users would be the Troubleshooting section on the same page.

Comment (DavidSCN) [reply Allwmake:41]: However, then let's go with this description.

**Pull request merged**